### PR TITLE
Add checkbox for displaying emojis in sidebar

### PR DESCRIPTION
**Pull request by nsarrazin**
https://github.com/huggingface/chat-ui/assets/25119303/98d15803-a37b-48e7-a539-90ace6ffccee

### CODE REVIEW

**Comment by nsarrazin** [comment]
I think it would be nicer if the form was blocking while applying the change to settings. 

Now it closes immediately and the changes happens in the background which makes UX a bit confusing, could be done later though. wdyt @gary149 ?

File: src/lib/components/SettingsModal.svelte [hunk 1]
```diff
@@ -75,6 +75,14 @@
 					</ul>
 				</div>
 			{/if}
+			<label class="flex cursor-pointer select-none items-center gap-2 text-sm text-gray-500">
+				<input
+					type="checkbox"
+					name="hideEmojiOnSidebar"
+					bind:checked={settings.hideEmojiOnSidebar}
+				/>
+				Hide emoji on sidebar
```
Comment (gary149) [inline src/lib/components/SettingsModal.svelte:84]: nit but i's not only is it in the sidebar, it's also in the top-level mobile navigation.

Comment (gary149) [reply src/lib/components/SettingsModal.svelte:84]: I think my proposed wording was fine: "Show emoticons in conversation topics"

File: src/routes/+layout.server.ts [hunk 2]
```diff
@@ -51,7 +51,7 @@ export const load: LayoutServerLoad = async ({ locals, depends, url }) => {
 			})
 			.map((conv) => ({
 				id: conv._id.toString(),
-				title: conv.title,
+				title: settings?.hideEmojiOnSidebar ? conv.title.replace(/\p{Emoji}/gu, "") : conv.title,
```
Comment (gary149) [inline src/routes/+layout.server.ts:54]: It's a bit weird to hide there? I would hide at bottom level (Svelte component)

**Comment by julien-c** [comment]
i haven't followed, but is there really a need for a end-user option for this?

**Comment by gary149** [comment]
> Now it closes immediately and the changes happens in the background which makes UX a bit confusing, could be done later though. wdyt @gary149 ?

I'm not too shocked by it. I think it's fine for now.

File: src/routes/+layout.server.ts (hunk 2 above)
Comment (nsarrazin) [reply src/routes/+layout.server.ts:54]: Thinking about this more, I think it makes more sense to keep it at the top level.

Both the page title, top-level mobile nav, and `NavConversationItem` use this layout data as a source of truth, so I would prefer to change it once at the top level than in three places.

File: src/lib/components/SettingsModal.svelte (hunk 1 above)
Comment (nsarrazin) [reply src/lib/components/SettingsModal.svelte]: Updated the wording!

**Pull request merged**